src/Lean/Data/Lsp/Ipc.lean

```diff
@@ -62,8 +62,20 @@ def readMessage : IpcM JsonRpc.Message := do
 def readRequestAs (expectedMethod : String) (α) [FromJson α] : IpcM (Request α) := do
   (←stdout).readLspRequestAs expectedMethod α
 
-def readResponseAs (expectedID : RequestID) (α) [FromJson α] : IpcM (Response α) := do
-  (←stdout).readLspResponseAs expectedID α
+/-- Reads response, discarding notifications in between (as this is purely for testing). -/
+partial def readResponseAs (expectedID : RequestID) (α) [FromJson α] :
```

Comment:
Could also be renamed if desired

Reply:
I'm O.K. with keeping the name if the doc comment describes why discarding notifications is desirable for testing.